app/actions/remote/custom_profile.ts

```diff
@@ -0,0 +1,145 @@
+ * @returns Promise with the custom profile attributes or error
+ */
+export const fetchCustomProfileAttributes = async (serverUrl: string, userId: string, filterEmpty = false): Promise<{attributes: CustomAttributeSet; error: unknown}> => {
+    const attributes: Record<string, CustomAttribute> = {};
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        let fields: CustomProfileField[] = [];
+        let attrValues: Record<string, string> = {};
+
+        try {
+            [fields, attrValues] = await Promise.all([
+                client.getCustomProfileAttributeFields(),
+                client.getCustomProfileAttributeValues(userId),
+            ]);
+        } catch (err) {
+            logDebug('error on fetchCustomProfileAttributes get fields and attr values', getFullErrorMessage(err));
+            return {attributes, error: err};
+        }
+
+        if (fields.length === 0) {
+            return {attributes, error: undefined};
+        }
+
+        try {
+            // Process each field to build attributes and collect promises
+            const attributeModelPromises: Array<Promise<Model[]>> = [];
+            const fieldModelPromises: Array<Promise<Model[]>> = [];
+            for (const field of fields) {
+                const value = attrValues[field.id] || '';
+                if (!filterEmpty || value) {
+                    attributes[field.id] = {
+                        id: field.id,
+                        name: field.name,
+                        value,
+                        sort_order: field.attrs?.sort_order,
+                    };
+
+                    attributeModelPromises.push(operator.handleCustomProfileAttributes({
+                        attributes: [{
+                            id: customProfileAttributeId(field.id, userId),
+                            field_id: field.id,
+                            user_id: userId,
+                            value,
+                        }],
+                        prepareRecordsOnly: true,
+                    }));
+                    fieldModelPromises.push(operator.handleCustomProfileFields({
+                        fields: [field],
+                        prepareRecordsOnly: true,
+                    }));
+                }
+            }
+
+            const [fieldResults, attributeResults] = await Promise.all([
+                Promise.all(fieldModelPromises),
+                Promise.all(attributeModelPromises),
+            ]);
+            const fieldBatch = fieldResults.flat();
+            const attributeBatch = attributeResults.flat();
+
+            if (fieldBatch.length > 0) {
+                await operator.batchRecords(fieldBatch, FIELD_HANDLER_DESCRIPTION);
+            }
+
+            if (attributeBatch.length > 0) {
+                await operator.batchRecords(attributeBatch, ATTRIBUTE_HANDLER_DESCRIPTION);
+            }
```

Comment:
I am still not completely convinced we need to do two different batches, but 0/5.